### PR TITLE
Merge website and wiki

### PR DESCRIPTION
**Pull request by rosano**
(continuing from #58)

This is a functional/deployable version of the site with [all the wiki content](https://wiki.remotestorage.io/Special:AllPages).

Live version: https://remotestorage.dev

See forum discussion: https://community.remotestorage.io/t/remotestorage-monthly-hangout-august-2021/708

- [x] change /CNAME https://github.com/remotestorage/website/pull/59#discussion_r696854184
- [x] change /docs/CNAME https://github.com/remotestorage/website/pull/59#discussion_r696854283
- [x] replace `contributors` with `organization_members` https://github.com/remotestorage/website/pull/59#discussion_r696961270

### CODE REVIEW

**Comment by raucao** [comment]
I went over the whole site and fixed the remaining broken wiki links, as well as improved some other things on the way.

I think for the contributors we can use the https://allcontributors.org bot to aggregate the list. It also allows manual additions, so this will then also allow us to add past and future non-code contributors that aren't part of any code commit history.

**Comment by rosano** [comment]
Much improved, great work.

I continue to believe the primary questions from most people who arrive here are "What apps can I use?" or "How do I integrate this into my project?". The protocol info isn't very convincing so I would have re-thought and re-written it before making it the top page. But as a navigation link I guess it doesn't get in anyone's way.

The sample code is great. Someday I will make a fiddle.

I learned about allcontributors recently. The idea is interesting but it feels like overkill to me as well. Seems like it should be possible to show organization members after all: https://github.com/remotestorage/website/pull/59#discussion_r696961270

I made a list of pre-merge todos in the first post: https://github.com/remotestorage/website/pull/59#issue-715354362
What else needs to be done? I'm too eager for people to look at something other than the current website.

**Comment by raucao** [comment]
> it should be possible to show organization members after all

That's cool. However, not all contributors are RS organization members, so the list would exclude some people. Maybe we start with that and then replace it with a script to add a custom list later.

**Comment by raucao** [comment]
>  I'm too eager for people to look at something other than the current website.

It's not very nice to dump on our past efforts like that continuously. I actually still think the current website *looks* nicer and has better layout for the explanations, even though I find the content organization and the addition of the wiki content a great improvement overall. There's no reason for you to keep telling me how much you hate the thing while at the same time asking for help.

**Comment by raucao** [comment]
> What else needs to be done?

We still need to re-organize the repo and make sure that the README is up to date:

> We will need to move things to the root directory of the repo eventually, and then also remove the old source code and instructions, and add new ones instead.

**Comment by rosano** [comment]
You're right, sorry about that. Will try to keep it positive.

I didn't touch `README.md` and `package.json` to maintain the setup information for the current website. Should I just purge everything else and just leave the newer information? Maybe also move out of `/docs`?

**Comment by raucao** [comment]
Yes, exactly. We can remove everything that isn't needed anymore and also move the pages out of `docs` to the root, because they're not really docs for the website. If we need something from the old site, we can simply get it from Git. It will also stay deployed as is until we change the DNS record.

We can test-drive with something like `new.remotestorage.io` first, and then change the main record when we're sure that everything works as intended.

**Comment by rosano** [comment]
Ok I'll work on that. If we're using GitHub Pages to deploy, I think the domain needs a `CNAME` for something like `rosano.github.io` and then configuration on the corresponding repo.

**Comment by rosano** [comment]
Ok I think I got my changes in here:
1) Use relative references for images
2) Point legacy redirects to static pages
3) Purge legacy site files and configuration
4) Move site from `/docs` to root
5) Document new installation in `README.md` and shortcuts in `package.json`

File: docs/index.md
Comment (rosano) [reply docs/index.md]: @raucao I realize I misinterpreted `organization_members` as "contributors from all projects in the organization", so yes this isn't so great. Eventually there should be a better proxy for contributors.

Comment (rosano) [reply docs/index.md]: or we can just take out the contributors list.

**Comment by raucao** [comment]
@rosano Great! I think we're close to merging and switching the site. :tada: 

I have set up GH Pages and deployed to https://remotestorage.dev so we can review the site on an apex domain (vs. subdomain).

I think switching domains entirely to `.dev` would probably lose us some Google juice, but tbh I'm not up to date with SEO at all. Maybe someone else has insights on if a permanent redirect (308) would impact it very much?

**Review by raucao — APPROVED** [review]

**Comment by raucao** [comment]
... added the org members as well now.

**Comment by rosano** [comment]
Looks great. I agree that changing the domain in production would probably cost some Google juice, not sure what the benefit would be.

When this is public I was planning on making some tweets to announce it, thank just-the-docs, and mention that edits are welcome. If you want to include anything else let me know so we can coordinate.

**Comment by raucao** [comment]
@remotestorage/contributors Could we get one more +1 to merge this and switch the production website to what you can see on https://remotestorage.dev right now?

@rosano Did you have an idea perhaps about what do with the header for the forums? Just replace the links there with the new ones for now I guess?

**Comment by rosano** [comment]
@raucao I just started to change them and then I thought it might be useful for a moment to leave it as it is to test redirects from the wiki to the new site.

**Comment by rosano** [comment]
@raucao BTW how should those wiki redirects be handled? I can re-create those half a dozen pages the same way as legacy ones like `/integrate` and then all you have to do is redirect the wiki domain to the site one.

**Comment by raucao** [comment]
They don't need to be the same URLs, because it's not the same domain.

My plan was to reconfigure the Discourse VM, so it doesn't route all 80/443 traffic to the Discourse container, but uses a separate Nginx on the host instead. Then we can add an nginx vhost config for `wiki.remotestorage.io` and redirect all the important pages (like Apps, Servers, etc.) directly to the new URLs, and all others simply to the frontpage.

File: wiki/developers.md
Comment (almereyda) [inline wiki/developers.md:74]: This section can be removed, if not written yet.

Else a single sentence can already give a hint, why this headline is mentioned here.

File: wiki/developers.md
Comment (almereyda) [inline wiki/developers.md:78]: Same as above.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:16]: This place intentionally left blank?

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:57]: Suggesting to use a whole line code block, either with prepending four spaces before the line, and leaving out the tick marks, or by wrapping the line into three-tick-marks-blocks.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:61]: Same as above.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:95]: This looks a little weird in the rendered output. Consider using a code block as with the previous comments.

    git clone https://github.com/remotestorage/remotestorage.js.git

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:112]: This link renders a little odd as well.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:168]: If mentioning the name after the three tick marks is to gain proper syntax highlighting, the string is appended directly after them, without space inbetween. This applies to all fenced code blocks.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:158]: This code block renders oddly enough to highlight it here. `html5` may not be a valid syntax highlighter for Jekyll. Try using `html` without the space to the annotation.

**Comment by raucao** [comment]
Alright, I have re-configured the Discourse VM, and added and tested an Nginx config for the wiki/redirects.

All ready for switching the domain (and then having to wait a bit to be able for DNS caches to update in order to get a Let's Encrypt cert for the wiki's new host).

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:230]: The next eleven lines are broken somehow. I don't know what's intended here, so I cannot comment on how fixing this.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:400]: Suggestion to use a fenced code block here, either with four spaces before the line, or three tick marks before and after.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:406]: The rendering of this line is broken. Consider removing the internal tick marks and `<>` characters, plus putting this in a code block.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:404]: Please add a simple explanatory paragraph/sentence to this section, or remove the stub.

File: wiki/beginners.md
Comment (almereyda) [inline wiki/beginners.md:6]: All these block comments appear a little off in the rendered output.

Screenshot 2021-08-30 at 19-31-25 Beginners

Please consider shortening them. The reference to how things were before can be omitted, because any new visitor to the site will not need to know that there was an old wiki in the first place.

Screenshot 2021-08-30 at 19-31-19 Beginners

This accounts for all of these blocks in the `wiki/` section.

**Review by almereyda — CHANGES_REQUESTED** [review]
Hi, @rosano asked me to leave a 👍🏽  on this PR, which made me review it along the way. After doing so, I've found some minor culprits which seem to arise from adapting the formatting of the wiki contents to the new environment.

I'd be happy to give this a go, if these occurrences can be eased out prior merging and going live. 

Additionally every page gives me

```
GET http://localhost:4000/favicon.ico
[HTTP/1.1 404 Not Found 0ms]
```

and `beginners`also adds an

```
GET http://localhost:4000/wiki/beginners/widget.png
[HTTP/1.1 404 Not Found 1ms]
```

**Comment by rosano** [comment]
I appreciate that you looked over the wiki pages. They were marked as 'in progress' to lighten the workload right now as it's years of edits by other people. I'll look at it but i'm not sure it's necessary to fix everything at the moment.

**Comment by almereyda** [comment]
I still have member access to the GitHub organisation, so could try adding the desired commits to this branch by myself, if you'd be okay with that.

It feels a little distracting to me to merge an incomplete change set, but I'd leave it to the maintainers to decide. What's your take at those typos above?

**Comment by rosano** [comment]
I pushed some changes:
1) Add `favicon.ico`
2) Remove outdated widget image reference
3) Use code fence blocks for better highlighting.

I decided to leave the placeholders as is because someone can at least search for `...` and write those parts. The objective of this pull request is simply to move over content as-is from the current website and wiki, not to modify or create much. Once it's live, anyone can more easily suggest changes of all sorts. Writing requires more time and thought.

**Comment by rosano** [comment]
It also says pretty clearly on the first line that it's in progress, so I would consider the current state acceptable.

File: wiki/beginners.md
Comment (rosano) [reply wiki/beginners.md:6]: I think this can be discussed in another pull request.

**Comment by raucao** [comment]
> Hi, @rosano asked me to leave a 👍🏽 on this PR, which made me review it along the way.

Thanks! :heart: :pray: 

> Also interesting to see that the @github-pages bot already deployed this branch, even well before it was merged.

That's because it currently deploys from this branch, so it's easy to review for everyone.

As for the Beginner's Guide, that was actually not even linked in the wiki anymore, because it has been superceded by https://remotestoragejs.readthedocs.io/en/latest/getting-started.html which didn't exist at that time. I'm going to just remove the page now.

**Review by almereyda — APPROVED** [review]
Thanks @rosano for including the fixes early on, despite they were well suited for a later, second stage editorial process. Incidentally, after removing that page altogether, the wiki section is in a much more concise state, with pages each holding similar amounts of information and more technically oriented contents.

This now looks very clean, and I'm looking forward to the subsequent rounds of editing, after this had been merged.

👍🏽

**Comment by raucao** [comment]
Aaand it's live! :tada: :rocket: 

I had a fun situation where the oldest website was actually being served at first, from https://github.com/remotestorage/remotestorage.io/ which still had the CNAME configured. Took me a minute to figure that out and fix it.

Great work, @rosano! And thanks to everyone else for the reviews! :clap:

**Comment by rosano** [comment]
Amazing, thanks everyone! I will announce on the rs social media accounts in a few hours and I guess I can also post on the forum.

I'll add assign myself an issue also for the contributor list reflecting commits.

**Pull request merged**
